### PR TITLE
[FIX] purchase: sections and notes.

### PR DESCRIPTION
**Pull request by Feyensv**
https://github.com/odoo/odoo/pull/34737 maybe didn't go in enough depth to 
ensure sections and notes don't break purchase flows.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase/models/purchase.py [hunk 1]
```diff
@@ -208,10 +208,9 @@ def onchange_partner_id(self):
     @api.onchange('fiscal_position_id')
     def _compute_tax_id(self):
         """
-        Trigger the recompute of the taxes if the fiscal position is changed on the PO.
+        Trigger the recomputation of the taxes if the fiscal position is changed on the PO.
         """
-        for order in self:
-            order.order_line._compute_tax_id()
```
Comment (Feyensv) [inline addons/purchase/models/purchase.py]: Maybe send this cosmetic change to master :D

File: addons/purchase/models/purchase.py [hunk 2]
```diff
@@ -504,15 +503,15 @@ def _prepare_compute_all_values(self):
         }
 
     def _compute_tax_id(self):
-        for line in self:
-            fpos = line.order_id.fiscal_position_id or line.order_id.partner_id.property_account_position_id
+        for line in self.filtered(lambda l: not l.display_type):
+            fpos = line.order_id.fiscal_position_id or line.order_id.partner_id.with_company(line.company_id).property_account_position_id
```
Comment (Feyensv) [inline addons/purchase/models/purchase.py]: Waiting for multi-company pr

**Comment by Feyensv** [comment]
@sle-odoo @pimodoo 

I recently thought back on the section and notes we added 3 months ago on the purchase orders and I think it might have broken some purchase flows :man_facepalming:.  

I'm fixing here what I saw as potentially bug-prone in the code, but I cannot ensure it fixes all the cases because I don't know those flows well...  Could you take a look at the purchase flows to ensure it works fine with section and notes ? 

Thanks.

**Comment by sle-odoo** [comment]
@Feyensv i think we can already merge this one and we'll ask our PO next week to play a little with the sections and purchase :)

**Comment by dbo-odoo** [comment]
indeed
@robodoo r+

**Pull request closed without merging**